File: 0205-isomorphic-strings/0205-isomorphic-strings.js
function isIsomorphic(s, t) {
    const len = s.length;
    const mapS = new Int32Array(256);
    const mapT = new Int32Array(256);

    for (let i = 0; i < len; i++) {
        const charS = s.charCodeAt(i);
        const charT = t.charCodeAt(i);

        if (mapS[charS] !== mapT[charT]) return false;

        mapS[charS] = i + 1;
        mapT[charT] = i + 1;
    }
    return true;
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        # Array size 256 acts as a fast primitive lookup structure for ASCII characters
        mapS = [0] * 256
        mapT = [0] * 256
        
        for i in range(len(s)):
            charS, charT = ord(s[i]), ord(t[i])
            
            if mapS[charS] != mapT[charT]:
                return False
                
            mapS[charS] = i + 1
            mapT[charT] = i + 1
            
        return True
